primitives/src/merkle_tree/mod.rs

```diff
@@ -80,53 +112,84 @@ where
     I: Index,
     T: NodeValue,
 {
-    // Possible improvement: adding digest_element() and digest_index()
     /// Digest a list of values
     fn digest(data: &[T]) -> T;
 
-    /// Digest a leaf (an indexed element)
+    /// Digest an indexed element
     fn digest_leaf(pos: &I, elem: &E) -> T;
 }
 
 /// An trait for Merkle tree index type.
-pub trait ToTraversalPath {
+pub trait ToTraversalPath<Arity: Unsigned> {
     /// Convert the given index to a vector of branch indices given tree height
     /// and arity.
-    fn to_traverse_path(&self, height: usize, arity: usize) -> Vec<usize>;
+    fn to_traversal_path(&self, height: usize) -> Vec<usize>;
```

Comment:
We could enforce the arity constraint at compile time by changing `Vec<usize>` to `GenericArray<usize, Arity>`

Reply:
Here the length is height + 1. Is it too complicated to make height as a trait as well?

Reply:
Oh right, the length of the output is not based on `Arity`. I was confused. I think making `height` a type parameter would be a good thing but might be more trouble than it's worth...currently we assume that a tree will be instantiated with a certain height at runtime and changing that to instantiate the type parameter at compile time might be troublesome. Never mind this comment, then